### PR TITLE
Run GPU python tests in the ROCm image

### LINKED ISSUE #140: Run GPU python tests in the ROCm image
To detect regressions in the ROCm packages (see #139).

### PR DESCRIPTION
**Pull request by jngrad**
Fixes #140

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -84,6 +84,7 @@ ubuntu-python3:cuda-9.0:build:
   extends: .build_template
 ubuntu-python3:cuda-10.1:build:
   extends: .build_template
+  timeout: 3h
```
Comment (mkuron) [inline .gitlab-ci.yml]: Why is this needed? I guess it shouldn‘t take longer than the other jobs.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -124,11 +125,21 @@ test/clang-python3:6.0:
...
+    - ctest -L gpu -LE long -E '^(tutorial|sample)_' --output-on-failure
   variables:
     make_check: "false"
     myconfig: "maxset"
     HIP_PLATFORM: "hcc"
     HCC_AMDGPU_TARGET: "gfx900"
+  tags:
+    - docker
+    - linux
+    - amdgpu
```
Comment (mkuron) [inline .gitlab-ci.yml]: Please set a 1h timeout. That‘s what we have in the main repo and running the tests does take a bit of time. The current 30m timeout was intended for compiling only.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -124,11 +125,21 @@ test/clang-python3:6.0:
...
+    - git clone --depth=1 --recursive https://github.com/espressomd/espresso
+    - cd espresso
+    - maintainer/CI/build_cmake.sh
+    - cd build
+    - ctest -L gpu -LE long -E '^(tutorial|sample)_' --output-on-failure
   variables:
     make_check: "false"
     myconfig: "maxset"
     HIP_PLATFORM: "hcc"
     HCC_AMDGPU_TARGET: "gfx900"
```
[the remark this answers is not in the document]
Comment (jngrad) [reply .gitlab-ci.yml]: Thanks!

File: .gitlab-ci.yml (hunk 1 above)
Comment (jngrad) [reply .gitlab-ci.yml]: master is currently red because 2h aren't always enough

File: .gitlab-ci.yml (hunk 2 above)
Comment (jngrad) [reply .gitlab-ci.yml]: I manually ran the test on coyote10  before opening the PR and it only took 4 min longer.

Comment (mkuron) [reply .gitlab-ci.yml]: Did you take into account that CI (but not manual testing) sets CPU limits and the machine was probably idle when you were testing?

Comment (jngrad) [reply .gitlab-ci.yml]: The CPU limit should not have too much of an impact since GPU python tests cannot run in parallel. But you're right, the number of other jobs running on the machine can have an impact. The timeout was increased to 1h.
